sql/core/src/test/scala/org/apache/spark/sql/DateFunctionsSuite.scala

```diff
@@ -475,6 +1164,45 @@ class DateFunctionsSuite extends QueryTest with SharedSQLContext {
       Row(ts1.getTime / 1000L), Row(ts2.getTime / 1000L)))
   }
 
+  test("to_unix_timestamp with session local timezone") {
```

Comment:
I don't think we need to add tests in this file at all. We should improve `DateTimeUtilsSuite` to make sure the newly added methods work well with different timezones, e.g. `getHours`, `daysToMillions`, etc. Then make sure these timezone aware expressions will call the newly added methods in `DateTimeUtils` which has timezone parameter(we can remove the old versions that don't take timezone parameter, after we finish handling partition values).

This suite is end-to-end test, and it's very annoying if we wanna test all changed expressions, we should write more low-level tests in `DateTimeUtilsSuite`.